Core/Systems/BarrierSystem/PlayerRadarBarrierHook.cs

```diff
@@ -28,161 +27,140 @@ private void DrawShieldForPlayers(On.Terraria.Main.orig_DrawInterface_14_EntityH
+					textString += " [c/64c8ff:" + mp.barrier + "/" + mp.maxBarrier + "]";
 
-					Vector2 pos = Main.MouseScreen + new Vector2(12f + vanillaTextPosition.X / 2 - textPosition.X / 2, vanillaTextPosition.Y);
-					if (pos.Y > Main.screenHeight - 30)
-						pos.Y = Main.screenHeight - 30;
-					if (pos.X > (float)(Main.screenWidth - textPosition.X))
-						pos.X = Main.screenWidth - textPosition.X;
+					if (eachPlayer.hostile)
+					{
+						textString += " " +  Terraria.Localization.Language.GetTextValue("Game.PvPFlag");
+					}
 
-					ChatManager.DrawColorCodedStringWithShadow(Main.spriteBatch, Terraria.GameContent.FontAssets.MouseText.Value, text, pos, 0f, Vector2.Zero, Vector2.One, out int hoveredSnippet);
+					Main.instance.MouseTextHackZoom(textString);
 				}
 			}
+			PlayerInput.SetZoom_UI();
 		}
 
-		/* TODO: replace this with new detour
-        private void DrawNewInfo(On.Terraria.Main.orig_DrawInterface_20_MultiPlayerPlayerNames orig)
-        {
-            orig();
-
-            var sb = Main.spriteBatch;
-
-            PlayerInput.SetZoom_World();
-
-            int width = Main.screenWidth;
-            int height = Main.screenHeight;
-            Vector2 screenPos = Main.screenPosition;
-
-            PlayerInput.SetZoom_UI();
-
-            float uiscale = Main.UIScale;
-
-            //Alot of this is vanillas code for finding the text position, I cant be assed to reverse engineer how they find where to draw the text.
-            for (int i = 0; i < Main.maxPlayers; i++)
-            {
-                if (Main.player[i].active && Main.myPlayer != i && !Main.player[i].dead && Main.player[Main.myPlayer].team > 0 && Main.player[Main.myPlayer].team == Main.player[i].team)
-                {
-                    var Player = Main.player[i];
-                    var mp = Player.GetModPlayer<Core.BarrierPlayer>();
-
-                    if (Player.statLife >= Player.statLifeMax2 && mp.Barrier >= mp.MaxBarrier)
-                        continue;
-
-                    string text = mp.Barrier + "/" + mp.MaxBarrier;
+		/// <summary>
+		/// Draw distant ally players' barrier on their healthbar
+		/// Using an IL because the On method seemed cursed
+		/// also this is more performant anyway
+		/// </summary>
+		/// <param name="il"></param>
+		private void DrawAllyRadarBarrierIL(ILContext il)
+		{
+			Type playerOffScreenCacheType = typeof(Mod).Assembly.GetType("Terraria.GameContent.UI.NewMultiplayerClosePlayersOverlay")!.GetNestedType("PlayerOffScreenCache", BindingFlags.NonPublic);
+			var playerAccessor = playerOffScreenCacheType.GetField("player", BindingFlags.NonPublic | BindingFlags.Instance)!;
+			var distanceDrawPositionAccessor = playerOffScreenCacheType.GetField("distanceDrawPosition", BindingFlags.NonPublic | BindingFlags.Instance);
 
-                    Vector2 textPosition = Terraria.GameContent.FontAssets.MouseText.Value.MeasureString(text);
+			StarlightRiver.Instance.Logger.Debug(distanceDrawPositionAccessor);
```

Comment:
Is there a reason to keep this logging beyond debugging/developing this? If not, we should probably remove it to avoid bloating logs